packages/markers/src/browser/problem/problem-composite-tree-node.ts

```diff
@@ -33,33 +33,46 @@ export namespace ProblemCompositeTreeNode {
         parent.severity = maxSeverity;
     };
 
-    export function addChild(parent: CompositeTreeNode, child: MarkerInfoNode, markers: Marker<Diagnostic>[]): CompositeTreeNode {
-        ProblemCompositeTreeNode.setSeverity(child, markers);
+    export function addChildren(parent: CompositeTreeNode, insertChildren: Map<string, { node: MarkerInfoNode, markers: Marker<Diagnostic>[] }>): void {
```

Comment:
Question: What is the significance of storing the children in a map, if we only access the values of the map in this function? IMO we should just use a plain array in that case.

Reply:
Array is slower than Map in some cases. But in this case, they take about the same amount of time. I will use array.

Reply:
> Question: What is the significance of storing the children in a map, if we only access the values of the map in this function? IMO we should just use a plain array in that case.

Array is slower than Map in some cases. But in this case, they take about the same amount of time. I will use array.